cas/ctg_store.c

```diff
@@ -268,23 +277,36 @@ int m0_ctg_create(struct m0_be_seg *seg, struct m0_be_tx *tx,
 		  struct m0_cas_ctg **out,
 		  const struct m0_fid *cas_fid)
 {
-	struct m0_cas_ctg *ctg;
-	int                rc;
+	struct m0_cas_ctg          *ctg;
+	int                         rc;
+	struct m0_btree_op          b_op    = {};
+	struct m0_btree_rec_key_op  key_cmp = { .rko_keycmp = ctg_cmp, };
+	struct m0_fid              *fid     = &M0_FID_TINIT('b',
+							cas_fid->f_container,
+							cas_fid->f_key);
+	struct m0_btree_type        bt      = {
+		.tt_id = M0_BT_CAS_CTG,
+		.ksize = -1,
+		.vsize = -1,
+	};
 
 	if (M0_FI_ENABLED("ctg_create_failure"))
 		return M0_ERR(-EFAULT);
 
-	M0_BE_ALLOC_PTR_SYNC(ctg, seg, tx);
+	M0_BE_ALLOC_ALIGN_PTR_SYNC(ctg, 12, seg, tx);
 	if (ctg == NULL)
 		return M0_ERR(-ENOMEM);
 
 	ctg_init(ctg, seg);
-	rc = M0_BE_OP_SYNC_RET(op,
-		       m0_be_btree_create(&ctg->cc_tree, tx, &op,
-					  &M0_FID_TINIT('b',
-							cas_fid->f_container,
-							cas_fid->f_key)),
-		       bo_u.u_btree.t_rc);
+	M0_ALLOC_PTR(ctg->cc_tree);
+	if (ctg->cc_tree == NULL)
```

Comment:
Free up memory allocated for `ctg` on line 296.

Reply:
it is not needed in this function, as ctg will get freed explicitly by caller of this function